controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook_test.go

```diff
@@ -209,11 +214,17 @@ func TestKubeadmControlPlaneValidateUpdate(t *testing.T) {
 	invalidUpdateKubeadmConfigInit := before.DeepCopy()
 	invalidUpdateKubeadmConfigInit.Spec.KubeadmConfigSpec.InitConfiguration = &kubeadmv1beta1.InitConfiguration{}
 
+	validUpdateKubeadmConfigInit := before.DeepCopy()
+	validUpdateKubeadmConfigInit.Spec.KubeadmConfigSpec.InitConfiguration.NodeRegistration = kubeadmv1beta1.NodeRegistrationOptions{}
+
 	invalidUpdateKubeadmConfigCluster := before.DeepCopy()
 	invalidUpdateKubeadmConfigCluster.Spec.KubeadmConfigSpec.ClusterConfiguration = &kubeadmv1beta1.ClusterConfiguration{}
 
+	invalidUpdateKubeadmConfigJoin := before.DeepCopy()
+	invalidUpdateKubeadmConfigJoin.Spec.KubeadmConfigSpec.JoinConfiguration = &kubeadmv1beta1.JoinConfiguration{}
```

Comment:
Assuming that join configuration initially has only  `JoinConfiguration.NodeRegistration`, 
https://github.com/kubernetes-sigs/cluster-api/blob/8a03ae7d608f6dfd3befa7b08ee076253d8b804e/controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook_test.go#L189-L191
this change now is valid.
Is it possible to Change the initial  `JoinConfiguration.NodeRegistration` so we actually test on invalid change and one valid change?

Reply:
It should have `JoinConfiguration.Discovery` set now - which should lead to this update being invalid.